dkc/core/models/file.py

```diff
@@ -67,10 +65,10 @@ def _file_pre_save(sender, instance, *args, **kwargs):
     instance.size = instance.blob.size
 
     if not instance.pk:  # this is a new file; reserve quota
-        instance.folder.increment_quota(instance.size)
+        instance.folder.effective_quota.increment(instance.size)
```

Comment:
I think until we see an implementation of the behavior of transferring a folder from a user quota to its own quota, this isn't an apples-to-apples comparison with my approach. I think that operation is likely to be one area where your proposed approach is less elegant.